packages/react-component-manifests/src/manifests/Icon/Icon.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { forwardRef, ReactNode, SVGProps, useCallback } from "react";
+import reactSchemaId from "@atrilabs/react-component-manifest-schema?id";
+import type { ReactComponentManifestSchema } from "@atrilabs/react-component-manifest-schema/lib/types";
+import iconSchemaId from "@atrilabs/component-icon-manifest-schema?id";
+import { CommonIcon } from "../CommonIcon";
+import CSSTreeId from "@atrilabs/app-design-forest/lib/cssTree?id";
+import { CSSTreeOptions } from "@atrilabs/app-design-forest/lib/cssTree";
+import { CustomPropsTreeOptions } from "@atrilabs/app-design-forest/lib/customPropsTree";
+import CustomTreeId from "@atrilabs/app-design-forest/lib/customPropsTree?id";
+
+export const Icon = forwardRef<
+  HTMLElement,
+  {
+    styles: React.CSSProperties;
+    custom: { svg: string };
+    onClick: (event: {
+      eventX: number;
+      eventY: number;
+      x: number;
+      y: number;
+      width: number;
+      height: number;
+    }) => void;
+    className?: string;
+  }
+>((props, ref) => {
+  const onClick = useCallback(
+    (e: React.MouseEvent) => {
+      const { x, y, width, height } = (
+        e.nativeEvent.target as HTMLElement
+      ).getBoundingClientRect();
+      props.onClick({
+        eventX: e.pageX,
+        eventY: e.pageY,
+        x,
+        y,
+        width,
+        height,
+      });
+    },
+    [props]
+  );
+  return (
+    <i
+      ref={ref}
+      className={props.className}
+      style={{
+        display: "inline-block",
```

Comment:
I've also removed `display: "inline-block"` since now I can set it in the initial value